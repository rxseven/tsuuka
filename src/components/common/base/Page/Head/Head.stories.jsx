import React from 'react';
import { storiesOf } from '@storybook/react';

import Head from './index';

const title = 'Components/Common/Base/Page/Head';
const component = (
  <Head>
    <title>Document Title</title>
    <meta content="Define a description of a web page" name="description" />
  </Head>
);

storiesOf(title, module).add('default', () => component, {
  notes: {
    markdown: `**No view layer** - This component will manage all of changes to the document head. It takes plain HTML tags and outputs plain HTML tags. For more information, see [rect-helmet](https://github.com/nfl/react-helmet).
    `
  }
});
