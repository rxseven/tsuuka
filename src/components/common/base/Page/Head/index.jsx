import React from 'react';
import { Helmet } from 'react-helmet';

function Head({ children }) {
  return <Helmet>{children}</Helmet>;
}

export default Head;
