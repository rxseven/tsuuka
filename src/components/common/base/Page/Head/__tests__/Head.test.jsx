import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Head from '../index';

describe('<Head />', () => {
  // Arrange
  const props = { children: 'content' };
  const component = <Head {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
