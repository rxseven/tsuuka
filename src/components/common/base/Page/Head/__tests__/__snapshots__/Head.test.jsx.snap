// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Head /> Snapshot tests should render correctly 1`] = `null`;
